### PR TITLE
remaining todos for unit tests from audit threat model

### PR DESCRIPTION
**Pull request by 0xAlcibiades**
Addressed all but 2 TODOs, which are related to possible contract changes:
- // TODO(Why do we need `claim.unredeemed` if claim() reverts when it does not exist?) See L64 below
- // TODO(Should fee sweep be onlyFeeTo? Probably)

### CODE REVIEW

File: test/OptionSettlementEngine.integration.t.sol [hunk 1]
```diff
@@ -242,44 +242,45 @@ contract OptionSettlementIntegrationTest is BaseEngineTest {
...
-        assertEq(claimRatios[1], bucketRatio1);
-        assertEq(claimRatios[2], bucketRatio1);
-
-        // Claims 4, 5, 6, and 7 are not exercised (0/7)
-        assertEq(claimRatios[3], bucketRatio2);
-        assertEq(claimRatios[4], bucketRatio2);
-        assertEq(claimRatios[5], bucketRatio2);
-        assertEq(claimRatios[6], bucketRatio2);
-    }
+    // TODO resolve test which broke after removing 0s from newOptionType
```
Comment (Flip-Liquid) [inline test/OptionSettlementEngine.integration.t.sol:245]: Can we either fix these in this PR or drop this commented code

**Review by Flip-Liquid — CHANGES_REQUESTED** [review]
Onky feedbakc is to fix the integration test before merge

File: test/OptionSettlementEngine.integration.t.sol (hunk 1 above)
Comment (neodaoist) [reply test/OptionSettlementEngine.integration.t.sol:245]: I looked into this failing test. It seems the different initial option key / settlement seed causes pretty significant changes to exercise assignment, I believe driven by L788 `uint96 exerciseIndex = uint96(optionRecord.settlementSeed % numUnexercisedBuckets);`. Raw bucket ratio outputs included below.

Taking a step back, what do we want to check w/ this test? What if we remove this test and write a new _unit/fuzz_ test that checks exercise assignment is fair and works as expected (notably missing from unit tests rn).

Before changing initial settlement seed (audit-fixes branch):

```
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 3000000000000000000
  claim ratio WAD        : 428571428571428571
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 2000000000000000000
  claim ratio WAD        : 285714285714285714
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 2000000000000000000
  claim ratio WAD        : 285714285714285714
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
```
  
After changing initial settlement seed (threat-model-tests branch):
  
```
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 4000000000000000000
  claim ratio WAD        : 571428571428571428
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 333333333333333333
  claim ratio WAD        : 47619047619047619
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 333333333333333333
  claim ratio WAD        : 47619047619047619
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 333333333333333333
  claim ratio WAD        : 47619047619047619
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 500000000000000000
  claim ratio WAD        : 71428571428571428
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 500000000000000000
  claim ratio WAD        : 71428571428571428
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 1000000000000000000
  claim ratio WAD        : 142857142857142857
```

Comment (0xAlcibiades) [reply test/OptionSettlementEngine.integration.t.sol:245]: @neodaoist let's just fix the integration test and get this merged. Consider R&D on the core contracts frozen in the new year.

Comment (neodaoist) [reply test/OptionSettlementEngine.integration.t.sol:245]: Agree 100%. I can't figure out how to fix this test tho, could use help from you or Flip. With the new settlement seed, all claims get some exercised from them. I'm not sure how to calculate what we expect each claim's ratio to be.

**Pull request merged**